sklearn/gaussian_process/kernels.py

```diff
@@ -0,0 +1,940 @@
+""" Kernels for Gaussian process regression and classification.
+
+The kernels in this module allow kernel-engineering, i.e., they can be
+combined via the "+" and "*" operators. These expressions can also contain
+scalar values, which are automatically converted to a constant kernel.
+
+All kernels allow (analytic) gradient-based hyperparameter optimization.
+The space of hyperparameters can be specified by giving lower und upper
+boundaries for the value of each hyperparameter (the search space is thus
+rectangular). This can be achieved by using a pair or triple instead of a
+single float wherever a parameter value is specified. In case of a pair,
+the first value specifies the lower boundary and the second value the upper
+boundary. In case of a triple, the middle value specified the initial value
+of the parameter during hyperparameter-optimization.
+"""
+
+# Author: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+# Licence: BSD 3 clause
+
+# Note: this module is strongly inspired by the kernel module of the george
+#       package.
+
+from abc import ABCMeta, abstractmethod
+from functools import partial
+import inspect
+
+import numpy as np
+from scipy.spatial.distance import pdist, cdist, squareform
+
+from ..metrics.pairwise import pairwise_kernels
+from ..externals import six
+from ..base import clone
+
+
+class Kernel(six.with_metaclass(ABCMeta)):
+    """ Base class for all kernels."""
+
+    def __init__(self, theta=1.0, thetaL=1e-5, thetaU=np.inf):
+        if not np.iterable(theta):
+            theta = np.array([theta])
+        self.theta = np.asarray(theta, dtype=np.float)
+        self.bounds = (np.asarray(thetaL, dtype=np.float),
+                       np.asarray(thetaU, dtype=np.float))
+
+    def get_params(self, deep=True):
+        """Get parameters of this kernel.
+
+        Parameters
+        ----------
+        deep: boolean, optional
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+        params = dict(theta=self.theta, thetaL=self.bounds[:, 0],
```

Comment:
One further alternative:
  * `__init__(sigma_1, sigma_2, sigma_1_bounds=(0, np.inf), sigma_2_bounds=(0, np.inf))`

Grouping the corresponding lower and upper bound together would be more concise and would allow to exclude hyperparameters from optimization by passing a special value like  `sigma_1_bounds="fixed"`.